### DISCUSSION TITLE
Twill 2.6.0

### DISCUSSION THREAD

**Discussion by ifox** [area17/twill#1328] [General]
Twill 2.6.0 is a non-breaking release bringing a lot of cool new features, fixes and improvements, especially for developers:

- we introduced new Artisan commands, and new options were added to existing commands:
	- `twill:make:singleton` will help you with standalone forms that can use all Twill modules features. They are a great replacement for sometimes lacking settings pages. 
	- the new `--hasNesting` option of the modules CLI will fully configure self-nested modules for you. No more snippets to copy and paste from the docs.
	- `twill:refresh-crops` can apply your new `mediasParams` configuration so that your crops are immediately available without using the CMS UI.

- to render images, we added support for [TwicPics](https://www.twicpics.com/). They are an amazing French-based alternative to Imgix that you can try through their [free plan](https://www.twicpics.com/pricing) which offers up to 3GB of CDN bandwidth, with unlimited master images and transformations.

- our documentation site has been separated into individual pages for better browsability and is now leveraging a couple more features from VuePress to improve collaboration. We hope this is a helpful change alongside all the new additions. More to come!

- Turkish translations were added, and German and Italian translations were updated by our amazing contributors!

- and a ton more fixes and improvements, check out the full changelog below!

Thank you to all our contributors and community members for their amazing support this year! We hope you all have a great end of 2021, see you in 2022! 🚀

**HOW TO UPDATE**

Run `composer update` in your project and then, run Twill's own update command: `php artisan twill:update`. This will force update your published Twill assets. You can delete the old ones from your repository if you are versioning them. If you are using custom Vue components, you should rebuild Twill assets using `php artisan twill:build`. 

### Added

- Artisan command `twill:make:singleton` to generate single-record modules [`#1178`](https://github.com/area17/twill/pull/1178)
- Option `--hasNesting` to generate self-nested modules [`#1140`](https://github.com/area17/twill/pull/1140) [`#1222`](https://github.com/area17/twill/pull/1222)
- Artisan command `twill:refresh-crops` to generate missing crops [`#1289`](https://github.com/area17/twill/pull/1289) [`8b1e4f6d`](https://github.com/area17/twill/commit/8b1e4f6d)
- TwicPics image service [`#1217`](https://github.com/area17/twill/pull/1217) [`ce15c4a5`](https://github.com/area17/twill/commit/ce15c4a5)
- Turkish language support [`#1134`](https://github.com/area17/twill/pull/1134)
- Support for translated permalinks in the title editor [`#1092`](https://github.com/area17/twill/pull/1092)
- Support for capsule service providers [`#1212`](https://github.com/area17/twill/pull/1212)
- Auto detect latitude-longitude value in location field [`#1275`](https://github.com/area17/twill/pull/1275) [`#1170`](https://github.com/area17/twill/pull/1170)
- Adds the ability to display an indexColumn selected from a relatedBrowser [`#1302`](https://github.com/area17/twill/pull/1302)

### Fixed

- Support attribute casting on model and translations with JSON field groups [`#1151`](https://github.com/area17/twill/pull/1151)
- Support dynamic repeater titles with JSON repeaters [`#1171`](https://github.com/area17/twill/pull/1171)
- Fix admin HTTP exception views detection [`#1213`](https://github.com/area17/twill/pull/1213)
- Prevent conflicts with built-in Vue component names [`#1164`](https://github.com/area17/twill/pull/1164)
- Add fallback to capsule model in permalink base [`#1216`](https://github.com/area17/twill/pull/1216)
- Add `doctrine/dbal` 3.0 support [`#1226`](https://github.com/area17/twill/pull/1226)
- Ensure capsule autoloading when config is cached [`#1242`](https://github.com/area17/twill/pull/1242)
- Fix edit link feature regression in Quill WYSIWYG [`#1270`](https://github.com/area17/twill/pull/1270)
- Remove references to deprecated Symfony class [`#1269`](https://github.com/area17/twill/pull/1269)
- Prevent undefined route errors in IconsController [`#1268`](https://github.com/area17/twill/pull/1268)
- Fix typo in `twill:capsule:install` command [`#1290`](https://github.com/area17/twill/pull/1290)
- Initialize undefined crops in cropper UI using first available ratio [`#1258`](https://github.com/area17/twill/pull/1258)
- Fix `byKey()` setting query when passing `section` argument [`#1303`](https://github.com/area17/twill/pull/1303)
- Remove duplicate test stub class [`#1311`](https://github.com/area17/twill/pull/1311)
- Fix destroy action on single nested items [`#1304`](https://github.com/area17/twill/pull/1304)
- Prevent multiple AJAX requests in Block Editor previews [`#1282`](https://github.com/area17/twill/pull/1282)
- Fix singleton routing for primary and secondary navigation support [`#1325`](https://github.com/area17/twill/pull/1325)
- Use case insensitive like operator in filterHandleTranslations for PostgresSQL support [`#1322`](https://github.com/area17/twill/pull/1322)

### Improved

- Update routes mapping order to allow overrides [`#1133`](https://github.com/area17/twill/pull/1133)
- Handle many-to-many relations in index columns [`#1174`](https://github.com/area17/twill/pull/1174)
- Add `capsule_repository_prefix` configuration [`#1209`](https://github.com/area17/twill/pull/1209)
- Support custom port in `dev_mode_url` configuration [`#1163`](https://github.com/area17/twill/pull/1163)
- Support additional table actions in module listing [`#1202`](https://github.com/area17/twill/pull/1202)
- Show red input count only if above 90% capacity [`#1237`](https://github.com/area17/twill/pull/1237)
- Collapse repeater blocks on page load [`#1296`](https://github.com/area17/twill/pull/1296)
- Update nested module count pluralisation [`#1251`](https://github.com/area17/twill/pull/1251)
- Support `titleKey` parameter in `relatedBrowsers` configuration [`#1301`](https://github.com/area17/twill/pull/1301)
- Update `HasSlug` to use Eloquent instead of DB facade [`#1309`](https://github.com/area17/twill/pull/1309)
- Update german translations [`#1235`](https://github.com/area17/twill/pull/1235)
- Update italian translations [`#1312`](https://github.com/area17/twill/pull/1312)
- Update docblock annotations [`#1167`](https://github.com/area17/twill/pull/1167/files)
- Update documentation [`#1165`](https://github.com/area17/twill/pull/1165) [`#1244`](https://github.com/area17/twill/pull/1244) [`#1236`](https://github.com/area17/twill/pull/1236) 
- Add pagination and a few more quality of life updates to the documentation [`#1131`](https://github.com/area17/twill/pull/1131)

### Chores

- Update npm dependencies in documentation [`377e5e0`](https://github.com/area17/twill/commit/377e5e0b27916861caa448ef899ea0e3fbeff648)
- Bump axios from 0.21.1 to 0.21.2 [`#1327`](https://github.com/area17/twill/pull/1327)

## New Contributors
* @MarcelWeidum made their first contribution in https://github.com/area17/twill/pull/1165
* @chrispymm made their first contribution in https://github.com/area17/twill/pull/1171
* @madalinignisca made their first contribution in https://github.com/area17/twill/pull/1133
* @afatmustafa made their first contribution in https://github.com/area17/twill/pull/1134
* @MamlukiSn made their first contribution in https://github.com/area17/twill/pull/1092
* @haringsrob made their first contribution in https://github.com/area17/twill/pull/1212
* @fabianfetting made their first contribution in https://github.com/area17/twill/pull/1235
* @avramch made their first contribution in https://github.com/area17/twill/pull/1244
* @freshleafmedia made their first contribution in https://github.com/area17/twill/pull/1251
* @bosix made their first contribution in https://github.com/area17/twill/pull/1296
* @neklein made their first contribution in https://github.com/area17/twill/pull/1304
* @AleksanderHideolab made their first contribution in https://github.com/area17/twill/pull/1312

**Full Changelog**: https://github.com/area17/twill/compare/2.5.2...2.6.0

<hr /><em>This discussion was created from the release <a href='https://github.com/area17/twill/releases/tag/2.6.0'>Twill 2.6.0</a>.</em>

**Comment by devimako** [comment]
How to add translations in singleton module after running the seeder getting a error 
Error:
Argument #1 ($values) must be of type array, bool given.

This is the seeder file after created a singleton module.
<img width="529" alt="Screenshot 2023-02-16 at 6 59 26 PM" src="https://user-images.githubusercontent.com/99637767/219377904-8565e08c-1d42-4d2f-8a70-0e09baf9aea0.png">

Can i know how to use translations in singleton module

Thanks,
